.github/workflows/ci.yml

```diff
@@ -38,130 +70,179 @@ jobs:
-    env:
-      RUSTFLAGS: "--cfg peniko_msrv"
+        os: [windows-latest, macos-14, ubuntu-latest]
     steps:
       - uses: actions/checkout@v4
 
-      - name: install msrv toolchain
+      - name: restore cache
+        uses: Swatinem/rust-cache@v2
+
+      - name: install stable toolchain
         uses: dtolnay/rust-toolchain@master
         with:
-          toolchain: ${{ env.MINIMUM_SUPPORTED_RUST_VERSION }}
-          components: clippy
+          toolchain: ${{ env.RUST_STABLE_VER }}
+
+      - name: cargo test
+        run: cargo test --workspace --locked --all-features
+
+  test-stable-wasm:
+    name: cargo test (wasm32)
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
 
-      - name: cargo clippy (no default features with libm)
-        run: cargo clippy --workspace --lib --bins --no-default-features --features libm -- -D warnings
+      - name: restore cache
+        uses: Swatinem/rust-cache@v2
 
-      - name: cargo clippy (no default features with libm) (auxiliary)
-        run: cargo clippy --workspace --tests --benches --examples --no-default-features --features libm -- -D warnings
+      - name: install stable toolchain
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: ${{ env.RUST_STABLE_VER }}
+          targets: wasm32-unknown-unknown
 
-      - name: cargo clippy (default features)
-        run: cargo clippy --workspace --lib --bins -- -D warnings -A clippy::doc_markdown
+      # TODO: Find a way to make tests work. Until then the tests are merely compiled.
+      - name: cargo test compile
+        run: cargo test --workspace --locked --target wasm32-unknown-unknown --all-features --no-run
 
-      - name: cargo clippy (default features) (auxiliary)
-        run: cargo clippy --workspace --tests --benches --examples -- -D warnings -A clippy::doc_markdown
+  check-msrv:
+    name: cargo check (msrv)
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [windows-latest, macos-14, ubuntu-latest]
+    steps:
+      - uses: actions/checkout@v4
 
-      - name: cargo clippy (all features)
-        run: cargo clippy --workspace --lib --bins --all-features -- -D warnings
+      - name: restore cache
+        uses: Swatinem/rust-cache@v2
 
-      - name: cargo clippy (all features) (auxiliary)
-        run: cargo clippy --workspace --tests --benches --examples --all-features -- -D warnings
+      - name: install msrv toolchain
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: ${{ env.RUST_MIN_VER }}
+          targets: x86_64-unknown-none
 
-      - run: rustup target add armv7a-none-eabi
+      - name: install cargo-hack
+        uses: taiki-e/install-action@v2
+        with:
+          tool: cargo-hack
 
-      # We use armv7a as a no_std with AtomicU64
-      - name: cargo build (no-default-features + libm, no_std target)
-        run: cargo build --no-default-features --features libm --target armv7a-none-eabi
```

Comment:
I believe we'll want to keep building this for the armv7a-none-eabi target, as it is a target without std, but that has the necessary target features to compile peniko.

Otherwise, it is possible that std can be pulled in unnoticed by a dependency and the testsuite succeeds with much of the same issue as feature unification.

Reply:
Nevermind, I see you switched that to x86-64-unknown-none now.